rolecall/rolecall.py

```diff
@@ -101,73 +46,166 @@ class RoleCall:
     def __init__(self, bot):
         self.bot = bot
         self.settings = dataIO.load_json(SETTINGS_PATH)
+        self.reaction_queue = {}
+        self.reaction_user_queue = set()
+        self.queue_processor_task = bot.loop.create_task(self.queue_processor())
+
+    def _record_entry(self, entry: Entry):
+        """ record entry to settings file """
+
+        server = self.settings[entry.server.id]
+        server.setdefault(entry.roleboard_channel.id, {})
+        server[entry.roleboard_channel.id].setdefault(entry.content_or_message_id, {})
+        keyring = server[entry.roleboard_channel.id][entry.content_or_message_id]
+        keyring[entry.emoji] = deepcopy(ROLE_RECORD_STRUCT)
+        keyring[entry.emoji]['ROLE_ID'] = entry.role.id
+        keyring[entry.emoji]['ROLE_NAME'] = entry.role.name
+        self._save()
+
+    def _check_entry(self, entry: Entry):
+        """ Checks if entry exists in the settings file and 
+        returns true if it does, otherwise, returns false.
+        """ 
+
+        entries = self.settings[entry.server.id][entry.roleboard_channel.id]
+        if entry.content_or_message_id in entries:
+            return True
+        else:
+            return False
+
+    async def _get_role_from_entry(self, entry: Entry):
+        """ Accesses board entry and retrieves role that corresponds 
+        to the emoji given """
+
+        server_id = entry.server.id
+        roleboard_id = entry.roleboard_channel.id
+        message_id = entry.content_or_message_id
+        keyring = self.settings[server_id][roleboard_id][message_id]
+        role_name = keyring[entry.emoji]['ROLE_NAME']
+        role = await self.get_or_create('role', role_name, entry.server)
+        return role
 
     @commands.group(pass_context=True, no_pm=True)
-    async def roleboard(self, ctx):
-        """change roleboard settings"""
+    async def rolecall(self, ctx):
```

Comment:
we should probably add more entry management before we publish.
doesn't need to be in-depth.